samples/nrf9160/gnss/src/assistance_minimal.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <zephyr.h>
+#include <sys/timeutil.h>
+#include <settings/settings.h>
+#include <logging/log.h>
+#include <nrf_modem_at.h>
+#include <nrf_modem_gnss.h>
+
+#include "assistance.h"
+#include "factory_almanac.h"
+#include "mcc_location_table.h"
+
+LOG_MODULE_DECLARE(gnss_sample, CONFIG_GNSS_SAMPLE_LOG_LEVEL);
+
+/* (6.1.1980 UTC - 1.1.1970 UTC) */
+#define GPS_TO_UNIX_UTC_OFFSET_SECONDS	(315964800UL)
+/* UTC/GPS time offset as of 1st of January 2017. */
+#define GPS_TO_UTC_LEAP_SECONDS		(18UL)
+#define SEC_PER_MIN			(60UL)
+#define MIN_PER_HOUR			(60UL)
+#define SEC_PER_HOUR			(MIN_PER_HOUR * SEC_PER_MIN)
+#define HOURS_PER_DAY			(24UL)
+#define SEC_PER_DAY			(HOURS_PER_DAY * SEC_PER_HOUR)
+#define DAYS_PER_WEEK			(7UL)
+#define PLMN_STR_MAX_LEN		8 /* MCC + MNC + quotes */
+
+static char almanac_checksum[64];
+
+static int set(const char *key, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	int len;
+	int key_len;
+	const char *next;
+
+	if (!key) {
+		return -ENOENT;
+	}
+
+	key_len = settings_name_next(key, &next);
+
+	if (!strncmp(key, "almanac_checksum", key_len)) {
+		len = read_cb(cb_arg, &almanac_checksum, sizeof(almanac_checksum));
+		if (len < sizeof(almanac_checksum)) {
+			LOG_ERR("Failed to read almanac checksum from settings");
+		}
+
+		return 0;
+	}
+
+	return -ENOENT;
+}
+
+static struct settings_handler assistance_settings = {
+	.name = "assistance",
+	.h_set = set,
+};
+
+static void factory_almanac_write(void)
+{
+	int err;
+
+	/* Check if the same almanac has already been written to prevent unnecessary writes
+	 * to flash memory.
+	 */
+	if (!strncmp(almanac_checksum, FACTORY_ALMANAC_CHECKSUM, sizeof(almanac_checksum))) {
+		LOG_INF("Factory almanac has already been written, skipping writing");
+		return;
+	}
+
+	err = nrf_modem_at_printf("AT%%XFILEWRITE=1,\"%s\",\"%s\"",
+				  FACTORY_ALMANAC, FACTORY_ALMANAC_CHECKSUM);
+	if (err != 0) {
+		LOG_ERR("Failed to write factory almanac");
+		return;
+	}
+
+	LOG_INF("Wrote factory almanac");
+
+	err = settings_save_one("assistance/almanac_checksum",
```

Comment:
Is this "assistance/almanac_checksum" setting written on the flash also or where is it stored? So the idea is to reduce the flash wear by writing the checksum there and checking against that if we need to update the (much larger) almanac on the flash?

Reply:
Yes. And the checksum is only written once, after the factory almanac has been written. After that the checksum is only read and compared, until the almanac in the sample is different and needs to be again updated.

Reply:
Oh yeah - the checksum is only written once (and next time only if it changes). I reviewed too quickly. Makes perfect sense now.